src/lint/linter/UberCheckstyleLinter.php

```diff
@@ -5,270 +5,338 @@
+        @$dom->loadXML($stdout);
+
+        $files = $dom->getElementsByTagName('file');
+        foreach ($files as $file) {
+            $errors = $file->getElementsByTagName('error');
+            $name = $file->getAttribute('name');
+            $arcPath = ltrim(str_replace(getcwd(), '', $name), '/');
+
+            if ($arcPath != $path) {
+                continue;
+            }
+
+            $changedLines = $this->getEngine()->getPathChangedLines($arcPath);
+            if ($changedLines) {
+                $changedLines = array_keys($changedLines);
+            } else {
+                $changedLines = array();
+            }
+
+            foreach ($errors as $error) {
+                $source = idx(array_slice(explode('.', $error->getAttribute('source')), -1), 0);
+                $line = $error->getAttribute('line');
+
+                // Do not fail for errors outside changed lines
+                $errorTypes = array(
+                    'JavadocTypeCheck',
+                    'JavadocMethodCheck',
+                    'RegexpSinglelineCheck'
+                );
+                if (in_array($source, $errorTypes) && !in_array($line, $changedLines)) {
+                    continue;
+                }
+
+                // checkstyle's XMLLogger escapes these five characters
+                $description = $error->getAttribute('message');
+                $description = str_replace(
+                    ['&lt;', '&gt;', '&apos;', '&quot;', '&amp;'],
+                    ['<', '>', '\'', '"', '&'],
+                    $description);
+
+                $message = id(new ArcanistLintMessage())
+                    ->setPath($path)
+                    ->setLine($line)
+                    ->setCode($this->getLinterName())
+                    ->setName($source)
+                    ->setDescription($description);
+
+                $column = $error->getAttribute('column');
+                if ($column) {
+                    $message->setChar($column);
+                }
+
+                $severity = $error->getAttribute('severity');
+                switch ($severity) {
+                    case 'error':
+                        $message->setSeverity(ArcanistLintSeverity::SEVERITY_ERROR);
+                        break;
+                    case 'info':
+                        $message->setSeverity(ArcanistLintSeverity::SEVERITY_ADVICE);
+                        break;
+                    case 'ignore':
+                        $message->setSeverity(ArcanistLintSeverity::SEVERITY_DISABLED);
+                        break;
+                    case 'warning':
+                    default:
+                        $message->setSeverity(ArcanistLintSeverity::SEVERITY_WARNING);
+                        break;
+                }
+
+                $this->addAutofixes($message, $source, $line, $path);
+                $this->addLintMessage($message);
+            }
+        }
     }
 
-    $file = new SplFileObject($path);
```

Comment:
Why are all these lines changing?

can you remove indentation changes from this change? It makes it harder to review than necessary